modules/fbs/fbs.module

```diff
@@ -639,35 +649,47 @@ function fbs_form_profile2_phone_validate($element, $form_state) {
 /**
  * On hold validation for fbs_form_profile2
  */
-function fbs_form_profile2_onhold_validate($element, $form_state) {
+function fbs_form_profile2_onhold_validate($element, &$form_state, $form) {
   $from_date = $form_state['input']['profile_provider_fbs']['field_fbs_on_hold']['und'][0]['from']['date'];
   $from_date_stamp = strtotime($from_date);
+  $default_from_date = $form['profile_provider_fbs']['field_fbs_on_hold']['und'][0]['from']['#default_value'];
+  $default_from_date_stamp = strtotime($default_from_date);
   $to_date = $form_state['input']['profile_provider_fbs']['field_fbs_on_hold']['und'][0]['to']['date'];
   $to_date_stamp = strtotime($to_date);
   $now = date('d-m-Y');
   $now_stamp = strtotime($now);
 
-  if ($from_date_stamp < $now_stamp) {
-    form_set_error('profile_provider_fbs][field_fbs_on_hold', t('Start date must be higher than or equal to today') . ' - ' . $now . '.');
+  if (!empty($default_from_date) && 
+    $from_date_stamp != $default_from_date_stamp && 
+    $from_date_stamp <= $now_stamp) {
+    form_set_error('profile_provider_fbs][field_fbs_on_hold', 
+      t('Your reservations is already on hold from start date: %default-date. If you wish to change the start date from: %default-date, the start date must be higher than or equal to today: %now', 
+      array(
+        '%default-date' => format_date($default_from_date_stamp, 'custom', 'd-m-Y'), 
+        '%now' => $now),
+      )
+    );
+  }
```

Comment:
I am not sure I understand the use case for this validation error.

It triggers if:
- The user already has reservations on hold
- The user sets from date to another date in the past

I would think that the current error message on 674 is sufficient.

Perhaps you think it is necessary because the changes on L589 reset the default values for the hold period so the user might not be aware of any current onhold values?

In that case I suggest (again) that we do not reset past on hold values and only validate the from date if it has changed from the current default.

That means:
- Removing lines 589-597
- Removing lines 662-672
- Changing the validation on line 673 to:

```php
if ((empty($default_from_date) || $default_from_date_stamp != $from_date_stamp) &&
  $from_date_stamp <= $now_stamp) {
```

Reply:
I think it makes good sense to clear the on_hold values, if the end_date is surpassed, since then the on_hold values are no longer effective. I have implemented the solution in respect to the client wishes - ultimately it must be the client decision if we want to clear the values or not.

If a user tries to alter the on_hold values (when they are set), the user is prompted to choose a start date larger than today. If the user input was made due to an (user) error, the user will have no way of resetting the selection if we do not provide the default_date entered. Then the user will either have to guess the old default date, or update the start date. Even though an update dont actually make any difference, it might confuse the user if we do not supply the opportunity to set the form back to its orginal state. Therefore we provide the default date in the validation string, to help the user set the field back to the original date, and hence validate the form.

Reply:
> If a user tries to alter the on_hold values (when they are set), the user is prompted to choose a start date larger than today. If the user input was made due to an (user) error, the user will have no way of resetting the selection if we do not provide the default_date entered. Then the user will either have to guess the old default date, or update the start date. Even though an update dont actually make any difference, it might confuse the user if we do not supply the opportunity to set the form back to its orginal state. Therefore we provide the default date in the validation string, to help the user set the field back to the original date, and hence validate the form.

Thanks for elaborating. I see your point. In that regard the improved feedback message makes sense.